lib/saml/settings_service.rb

```diff
@@ -33,14 +48,15 @@ def connection
       memoize :connection
 
       def metadata
+        @fetch_attempted = true
         attempt ||= 0
         response = connection.get
         raise SAML::InternalServerError, response.status if (400..504).cover? response.status.to_i
         response.body
       rescue StandardError => e
         attempt += 1
         msg = "Failed to load SAML metadata: #{e.message}: try #{attempt} of #{METADATA_RETRIES}"
-        attempt >= METADATA_RETRIES ? Rails.logger.error(msg) : Rails.logger.warn(msg)
+        attempt >= METADATA_RETRIES ? log_message_to_sentry(msg, :error) : log_message_to_sentry(msg, :warn)
```

Comment:
Why not use an if/else here rather than a ternary?

```
if attempt >= METADATA_RETRIES
  log_message_to_sentry(msg, :error)
else
  log_message_to_sentry(msg, :warn)
  sleep attempt * 0.25 
  retry
end
```